tests/contrib/funsor/conftest.py

```diff
@@ -0,0 +1,13 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+
+def pytest_collection_modifyitems(items):
+    for item in items:
+        if item.nodeid.startswith("tests/contrib/funsor"):
+            if "stage" not in item.keywords:
+                item.add_marker(pytest.mark.stage("funsor"))
+            if "init" not in item.keywords:
+                item.add_marker(pytest.mark.init(rng_seed=123))
```

Comment:
I'm curious whether and how pytest avoids import errors when collecting tests on a system on which funsor is not installed. Do we want to support e.g. `make test` which runs only `stage=unit` (no funsor tests) on a system when funsor is not installed?

One possible solution is, similar to my tests using BioPython,
```python
try:
    import funsor
except ImportError:
    pytestmark = pytest.mark.skip(reason="funsor is not installed")
```

Reply:
I tried something like that, we'll see if it works.